metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py

```diff
@@ -24,9 +26,36 @@ class StructuredLinkableSpecName:
 
     entity_link_names: Tuple[str, ...]
     element_name: str
-    time_granularity: Optional[TimeGranularity] = None
+    time_granularity_name: Optional[str] = None
     date_part: Optional[DatePart] = None
 
+    def __post_init__(self) -> None:
+        """Post init method to ensure all name values are lower-case.
+
+        This implies that all values in the TimeGranularity enumeration must be lower-case.
+        """
+        assert self.element_name.lower() == self.element_name, (
```

Comment:
This means that custom granularities will be required to be lowercase. Is that a necessary restriction?

Reply:
In my opinion, yes. We do not want `Fortnight` and `fortnight` to mean different things in the same model. But also this this the standard we've used for ages.